UpgradeGuide.md

```diff
@@ -0,0 +1,73 @@
+# Upgrade Guide: Amplify Library for Swift
+
+As part of the latest version of Amplify Library for Swift, we have re-written our APIs to support idiomatic Swift features. 
+
+### Structured Concurrency (async/await pattern)
+
+All our APIs have been refactored to follow the async/await structured concurrency pattern. The following is an example of how you would implement Sign-In for your authentication flows:
+
+v1 - Callback api
+
+```
```

Comment:
Maybe add `swift` to all code blocks so we get syntax highlight? 

e.g. having `swift`:
```swift
func signIn(username: String, password: String) {
    Amplify.Auth.signIn(username: username, password: password) { result in
        switch result {
        case .success:
            print("Sign in succeeded")
        case .failure(let error):
            print("Sign in failed \(error)")
        }
    }
}
```

vs without:

```
func signIn(username: String, password: String) {
    Amplify.Auth.signIn(username: username, password: password) { result in
        switch result {
        case .success:
            print("Sign in succeeded")
        case .failure(let error):
            print("Sign in failed \(error)")
        }
    }
}
```

Reply:
updated